src/Database/Database.php

```diff
@@ -1512,16 +1512,18 @@ public function count($table, array $condition = [])
 	public function toArray($stmt, $do_close = true)
 	{
 		if (is_bool($stmt)) {
-			return $stmt;
```

Comment:
I would vote for moving this change into a separate PR, since it changes functionality and we have to make sure that it doesn't break anything which will cost some time - while the main issue is unfixed.

Reply:
On the contrary, we just have been lucky all this time. I checked  `DBA::toArray` usage and we systematically expect an array. This means no query have been erroring out, mostly because we use the `DBA::select` syntax as well that leaves little margin to make a SQL mistake.